orkes-conductor-queues/src/main/java/io/orkes/conductor/queue/config/RedisQueueConfiguration.java

```diff
@@ -12,31 +12,30 @@
  */
 package io.orkes.conductor.queue.config;
 
-import java.util.HashSet;
-import java.util.List;
-import java.util.Set;
-import java.util.stream.Collectors;
-
-import org.apache.commons.pool2.impl.GenericObjectPoolConfig;
-import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty;
-import org.springframework.boot.context.properties.EnableConfigurationProperties;
-import org.springframework.context.annotation.Bean;
-import org.springframework.context.annotation.Configuration;
-import org.springframework.context.annotation.Primary;
-
 import com.netflix.conductor.core.config.ConductorProperties;
 import com.netflix.conductor.dao.QueueDAO;
-
+import io.micrometer.core.instrument.MeterRegistry;
 import io.orkes.conductor.queue.dao.ClusteredRedisQueueDAO;
 import io.orkes.conductor.queue.dao.RedisQueueDAO;
-
-import io.micrometer.core.instrument.MeterRegistry;
 import lombok.extern.slf4j.Slf4j;
+import org.apache.commons.pool2.impl.GenericObjectPoolConfig;
+import org.springframework.boot.autoconfigure.AutoConfiguration;
+import org.springframework.boot.autoconfigure.EnableAutoConfiguration;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Import;
+import org.springframework.context.annotation.Primary;
 import redis.clients.jedis.*;
 
-@EnableConfigurationProperties(QueueRedisProperties.class)
-@Configuration
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+@EnableAutoConfiguration
+@AutoConfiguration
 @Slf4j
+@Import({io.orkes.conductor.queue.config.QueueRedisProperties.class})
```

Comment:
I could not make it work with any other way of autowiring configuration.
Please feel free to make any changes if you have a more elegant way to add these properties

Reply:
I think the library is missing a clear concept of how to integrate with a Spring / Spring Boot application. Ideally, there is a module that contains the library features and a second module that provides a Spring auto-configuration that activates automatically when added as a dependency to a Spring project. This is not the case here. Instead, there are some `@Configuration` and `@ConfigurationProperties` classes but no clear guidance of how to activate them.

In the README, it says somewhere that you should add a `@ComponentScan("io.orkes.conductor")` to your code. Did you do that? 

Anyway, medium term I suggest to add a separate "*-spring" module with auto-configurations. Short term I suggest to provide clear guidance of how to integrate this library with a Spring Boot application.

Reply:
@ComponentScan("io.orkes.conductor")
This is done in the conductor repository - where we load the application library
`@ComponentScan(basePackages = {"com.netflix.conductor", "io.orkes.conductor"})
`